common/system.h

```diff
@@ -326,6 +326,36 @@ class OSystem : Common::NonCopyable {
 	 */
 	virtual void engineDone() { }
 
+	/**
+	 * Ixdentify a task that ScummVM can perform.
+	 */
+	enum Task {
+		/**
+		 * The local server is running, allowing connections from other devices to transfer files.
+		 */
+		kLocalServer,
+
+		/**
+		 * ScummVM is downloading games or sync'ing savegames from the cloud.
+		 */
+		kCloudDownload,
```

Comment:
Do we need different tasks for CloudDownload and DataPackDownload? Or could we unify them into kRemoteServer meaning a ScummVM connection to a remote server?

Reply:
I could also remove the Task enum entirely as I don't currently need that information. But I though that in the future backends may need it. The current granularity is based on the implementation. I for example initially intended to separate `kCloudSavegameSync` and `kCloudGameDownload`. But it was easier in the cloud code to group the two. On the other end the data pack download uses a completely different code so is trivial to separate. But I could indeed easily group it with the CloudDownload in this enum if the separation is deemed useless.